### PR TITLE
UCT/IB/MLX5: A few fixes

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: src/uct/ib/mlx5/dv/ib_mlx5dv_md.c [hunk 1]
```diff
@@ -649,6 +649,12 @@ static ucs_status_t uct_ib_mlx5dv_md_open(struct ibv_device *ibv_device,
         goto err_free;
     }
 
+#if HAVE_DECL_IBV_ODP_SUPPORT_IMPLICIT
+    if (dev->dev_attr.odp_caps.general_caps & IBV_ODP_SUPPORT_IMPLICIT) {
+        dev->flags |= UCT_IB_DEVICE_FLAG_ODP_IMPLICIT;
```
Comment (yosefe) [inline src/uct/ib/mlx5/dv/ib_mlx5dv_md.c]: should we get rid of uct_ib_device_odp_has_global_mr() and make some common function to be used by uct_ib_verbs_md_open() as well?

File: test/gtest/uct/test_stats.cc [hunk 2]
```diff
@@ -271,6 +271,7 @@ UCS_TEST_SKIP_COND_P(test_uct_stats, put_short,
     EXPECT_STAT(sender, uct_ep, UCT_EP_STAT_PUT, 1UL);
     EXPECT_STAT(sender, uct_ep, UCT_EP_STAT_BYTES_SHORT,
                 sizeof(send_data));
+    flush();
```
Comment (yosefe) [inline test/gtest/uct/test_stats.cc]: why is it needed?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11859/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 19-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/e5179d3cad66fd52b3361aaceee3397e848dafc4/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/9a863c4300373248071a47d9655db622dd475084/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/27f950ffbf2f33627388ce32f59bcaedf89b01be/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/ff77121a1aa460aa6b5b66df7477bc62e0c226ba/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/a3731c0ecfd46bae6a7261417835bec13a5c73c3/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/4bca14f18f11fce3b714470ff6a71e253d1a9083/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/35ea26bb90dc8cb1ea2ecea36a9eef94746b9677/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/51e3f92dbb63c0ae0763286ed7b801045a0ab3e2/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/1f143f47cbe09d6e81b5d4dccd783ec5ebf62116/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/f2d4a50b50a014b0157a9ec4ee854e1a3f66d279/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/3b028327c70c91d299fb8ec17390fe521969be6a/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/c9b55a80eb0f256a1f75a5c5d1f4dd1a06d35421/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/c40131de217d88a399993d6a4673e662acfcd9fb/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/4bb47ac66ef1e4cc6ecdaef4559d4ff66154c950/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/276f4ceb75c40886dca57a0f0769d02cd96368c6/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/13515af5f7e91ce65258bd6c72f81b8c6bddd560/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/aadcf26b1ec293dbcebd04bd29b6c75e61614b98/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/5b4f523831fdbadb56dc970b92045f78c734efa3/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/f57c8fee0ad7e5ea4a9c3579f93f6b49ab5b427d/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/d5ce31b88bf90c798af49c85fb3973d1a7f84a6d/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/9f7d56239f337342ab06c19d22cee2cf261ffb8d/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/7ca57e2af3dc8d45510112832056596c1d84476c/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/2a7a35095d1cdbf3ac72f9b48eb930552e8c20b7/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/f2b0b51d7d5af684c2e64ddcf24dffbaafd80a91/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/72705a148e48c07d037aee78659ffe781bd32137/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/c3cebcf42e13cde4c4cf246349193f8f63128707/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/66d4c95359e943514d7b5140f1d3c9034376485c/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/cdf654903f6c0cfba5e2ecc7f4da2d0feaa6fc1d/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/936f7a73cdaebfad80f836b66c9b8d23/raw/a1c3ac01c427d612fde87c6d34fe24f29c2912b2/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11868/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 20-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/475dc4a988122f39aa60d9727622eb0659759f94/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/1d824652206a67ab838e581c13a9cecda742b25d/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/ac8269382e1d830e8aed091cf02fc08cb675af81/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/7d694577eb78ab645ebef0c605af8668da28dba6/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/65039551b338848ba5d33ffd9a81b934371ff5e8/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/139cbe1b6d2932c2f7d468e959c2b9e7a1d7c4c7/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/e258ca8cf5bd01c6df03886598f3d4df729256db/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/0d25b7f853e91e108dffc920822eb9fd3596b738/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/fa63fddb40b1da96492d52582416e19c622becd7/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/7bf24c164b0c7aa7178ec7d1f3394684cbfe5044/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/3f85c3acbde673b01aca52a0970908cf3e1755ee/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/85a714484d6b469c92ac555efcb8be85e96dbe95/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/52116bad4b57af11b3aac7896ffcaaf4471a3789/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/a80d64a53f09632a3e6bb9ce2fba70310ae7472d/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/9a5261e262446e77ca5c7625e64c43b8e3142184/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/b44d5d41c1f262b200b02c4012f27b5a98c725c6/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/2314af591457f1e7416778afb286044bcb398f8b/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/7504e689446f017451a6aad24c6c24cd62d44fca/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/a4272d728cb9a0082c46e0c1d20ad407034aeffe/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/fe84bbde5a8041f08cf9ab992cc4f36e834e2666/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/d922ce18a5d7c33d5b2a4226afd2e1446dcfab5e/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/11394fb461b40dcc7e4cfd657c6890dd11b3ce73/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/910995bcbc5a668329209fcb11e73a22570a2878/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/634eaaa6bc382783a7f457450ece866f3c89b310/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/c9bab6fa87765800cb3fbffc80a77f86475fcdb5/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/42d6400491190c1cd8fb655f8987c294482ee6a4/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/443a63b9226cee7c908414f05f90afb73a15dd0f/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/15533932f609481983de71348866785add327b1f/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/37167510c42284045c7ca81fedda5216/raw/ae87aa89af713ea035a1c817778d257e4a0c116a/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**